src/clinical/api/clinical-api.ts

```diff
@@ -186,15 +183,39 @@ class ClinicalController {
 
 		const donorEntityData = await service.getDonorEntityData(donorIds);
 
+		const entityRecords = donorEntityData[0].records;
+
+		// File Table can request multiple programs
+		const donorPrograms = entityRecords
+			.map((record) => record.program_id as string)
+			.filter((programId, index, array) => programId && array.indexOf(programId) === index);
+
 		const date = currentDateFormatted();
 		const fileName = `filename=Donor_Clinical_Data_${date}.zip`;
+
+		const exceptions = await Promise.all(
+			donorPrograms.map(async (programShortName) => {
+				const programDonors = entityRecords
+					.filter((record) => record.donor_id && record.program_id === programShortName)
+					.map((record) => record.donor_Id as number);
+
+				const programExceptions =
+					(await getExceptionManifestRecords(programShortName, {
+						donorIds: programDonors,
+						submitterDonorIds: [],
+					})) || [];
+
+				return programExceptions;
+			}),
+		).then((exceptionsArray) => exceptionsArray.flat());
+
 		res
 			.status(200)
 			.contentType('application/zip')
 			.attachment(fileName)
 			.setHeader('content-disposition', fileName);
 
-		const zip = createClinicalZipFile(donorEntityData);
+		const zip = createClinicalZipFile(donorEntityData, exceptions);
```

Comment:
This reads very nicely now. Zip file will contain donor entity data and exceptions. Looks great.